src/js/component.js

```diff
@@ -1425,13 +1425,18 @@ class Component {
    * Registers a component
    *
    * @param {String} name Name of the component to register
-   * @param {Object} comp The component to register
+   * @param {Object} ComponentToRegister The component to register
    * @static
    * @method registerComponent
    */
-  static registerComponent(name, comp) {
-    if (!name) {
-      return;
+  static registerComponent(name, ComponentToRegister) {
+    if (typeof name !== 'string' || !name.length) {
+      throw new Error(`illegal component name, "${name}"; must be a non-empty string`);
```

Comment:
Actually, no, it shouldn't. It will only get to the second part of the expression if it's a string.